src/main/kotlin/sk/streetofcode/webapi/service/YouTubeServiceImpl.kt

```diff
@@ -0,0 +1,56 @@
+package sk.streetofcode.webapi.service
+
+import com.google.api.services.youtube.YouTube
+import com.google.api.services.youtube.model.Video
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import sk.streetofcode.webapi.api.YouTubeService
+import sk.streetofcode.webapi.client.youtube.YoutubeProperties
+
+@Service
+class YouTubeServiceImpl(val youtube: YouTube, val youtubeProperties: YoutubeProperties) : YouTubeService {
+    companion object {
+        private val log = LoggerFactory.getLogger(YouTubeServiceImpl::class.java)
+    }
+
+    private fun getVideosById(id: List<String>): List<Video> {
+        val request = youtube.Videos().list(listOf("id", "snippet"))
+        request.id = id
+        return request.execute().items
+    }
+
+    private fun getPlaylistVideos(playlistId: String): List<Video> {
+        val videoIdList: MutableList<String> = ArrayList()
+
+        val playlistItemRequest = youtube.playlistItems().list(listOf("contentDetails"))
+        playlistItemRequest.fields = "nextPageToken,items/contentDetails/videoId"
+        playlistItemRequest.maxResults = 50
+        playlistItemRequest.playlistId = playlistId
+
+        var nextToken: String? = null
+        do {
+            playlistItemRequest.pageToken = nextToken
+            val playlistItemResult = playlistItemRequest.execute()
+            for (playlistItem in playlistItemResult.items) videoIdList.add(playlistItem.contentDetails.videoId)
+            nextToken = playlistItemResult.nextPageToken
+        } while (nextToken != null)
+
+        val videos = videoIdList.chunked(50).map { chunk ->
+            getVideosById(chunk)
+        }.flatten()
+
+        return videos
+    }
+
+    override fun getVideos(): List<Video> {
```

Comment:
I know that everything works fine and Elon Musk would hate me but I am thinking about simple optimalization. 
1. HashMap of <id, video> which will be populated on the first getVideos and afterwards getVideos would not try to fetch from the API
2. getVideoById would return from hashMap
3. if hashMap does not contain slug, then create one request as is now getVideosById, if slug/id truly exists, then it would also be added to hashMap

This process would speed up built of videa pages by  ..... :( I would tell you precise number but quota ended and we need new api key :D

Reply:
After your last commit:
├ ● /videa (ISR: 600 Seconds) (543 ms)                           2.44 kB         284 kB
└ ● /videa/[slug] (3508 ms)                                      3 kB            285 kB
    ├ /videa/UwWYdC16nko
    ├ /videa/04ttShc952U
    ├ /videa/HQkGx5kOA4k
    
Before adding cache:
├ ● /videa (ISR: 600 Seconds) (3554 ms)                               2.44 kB         284 kB
└ ● /videa/[slug] (10749 ms)                                          3 kB            285 kB
    ├ /videa/UwWYdC16nko (391 ms)
    ├ /videa/NTepSW7YrJs (338 ms)
    ├ /videa/2B8-78Re42I (323 ms)

Reply:
Cache added in https://github.com/StreetOfCode/streetofcode-web-api/pull/17/commits/166170f69f93e2b15afd8b9c774c7b2cc2143d8c. Seems to work OK, let's keep it. Hopefully no bugs will emerge 🤞